### PR TITLE
Pulumi 3.0 Upgrade

### PR DESCRIPTION
**Pull request by tjholm**
Have done base requirements a pulumi 3 upgrade with some basic manual testing. Have created NIT-319 for a broader refactor of our deployment functionality, this should also including mocking and unit testing of all of this functionality.

### CODE REVIEW

File: packages/plugins/aws/src/tasks/deploy/index.ts [hunk 1]
```diff
@@ -93,7 +92,7 @@ export class Deploy extends Task<void> {
 						}
 					} catch (e) {
 						fs.appendFileSync(errorFile, e);
-						pulumi.log.error(e);
+						pulumi.log.error('There we an error deploying the stack please check error logs for more detail');
```
Comment (jyecusch) [inline packages/plugins/aws/src/tasks/deploy/index.ts:95]: Should we link to the logs location or a docs page to make finding the logs easier?

File: packages/plugins/aws/src/tasks/deploy/index.ts [hunk 2]
```diff
@@ -114,7 +113,9 @@ export class Deploy extends Task<void> {
 				this.update(changes);
 			}
 		} catch (e) {
-			console.log(e);
+			fs.appendFileSync(errorFile, e);
+			throw new Error('An error occurred during deployment, please see latest aws:error log for more details');
```
Comment (jyecusch) [inline packages/plugins/aws/src/tasks/deploy/index.ts:117]: Same as the previous comment

File: packages/plugins/gcp/src/commands/deploy/gcp.ts [hunk 3]
```diff
@@ -8,30 +8,31 @@ import inquirer from 'inquirer';
...
+	// 'asia-south1', // Api gateway not supported
+	// 'asia-southeast1', // Api gateway not supported
+	// 'asia-southeast2', // Api gateway not supported
+	// 'asia-east2', // Api gateway not supported
 	'asia-east1',
-	'asia-northeast1',
-	'asia-northeast2',
-	// 'asia-northeast3',
-	'asia-southeast1',
+	// 'asia-northeast1', // Api gateway not supported
```
Comment (jyecusch) [inline packages/plugins/gcp/src/commands/deploy/gcp.ts:31]: Should these only be excluded (or warned and confirmed) when the feature (API gateway this time) isn't supported for the current stack? Not something to change here, but maybe a good idea for the backlog and discussion

File: packages/plugins/gcp/src/tasks/deploy/index.ts [hunk 4]
```diff
@@ -76,7 +76,7 @@ export class Deploy extends Task<void> {
 							createEntrypoints(stack.getName(), entrypoints, deployedSites, deployedApis, deployedFunctions);
 						}
 					} catch (e) {
-						pulumi.log.error(e);
+						pulumi.log.error('An error occurred, see latest gcp:error log for details');
```
Comment (jyecusch) [inline packages/plugins/gcp/src/tasks/deploy/index.ts:79]: Same comment on finding logs

Comment (tjholm) [reply packages/plugins/gcp/src/tasks/deploy/index.ts:79]: Have created NIT-329 to address this

File: packages/plugins/gcp/src/commands/deploy/gcp.ts (hunk 3 above)
Comment (tjholm) [reply packages/plugins/gcp/src/commands/deploy/gcp.ts:31]: Have created NIT-330 to address this

File: packages/plugins/aws/src/tasks/deploy/index.ts (hunk 2 above)
Comment (tjholm) [reply packages/plugins/aws/src/tasks/deploy/index.ts:117]: Have created NIT-329 to address this

File: packages/plugins/aws/src/tasks/deploy/index.ts (hunk 1 above)
Comment (tjholm) [reply packages/plugins/aws/src/tasks/deploy/index.ts:95]: Have created NIT-329 to address this

**Review by jyecusch — APPROVED** [review]

**Pull request merged**